### ISSUE TITLE
"Path.append called with non-local second path" while building with --build-dir

### ISSUE THREAD

**Issue by andreypopp** [ocaml/dune#979]
I've tried to use `dune` with `esy` configured to build directly into esy's build store and got this error:
```
% esy build
info esy build 0.1.27
info Building @opam/dune@1.0.0: starting
info Building @opam/dune@1.0.0: complete
Internal error, please report upstream including the contents of _build/log.
Description:
("Path.append called with non-local second path"
 (a (In_build_dir .sandbox/3270848ae00e988d779746c0f86bd0e4))
 (b (In_build_dir default/lib/.merlin-exists)))
Backtrace:
Raised at file "src/dep_path.ml", line 45, characters 32-55
Called from file "src/fiber/fiber.ml", line 243, characters 6-18
esy-build-package: command
                   failed:
                   '/Users/andreypopp/Workspace/esy-ocaml/esy-ocaml-project/node_modules/.cache/_esy/store/i/opam__slash__dune-1.0.0-1d39b004/bin/dune'
                   'build'
                   '--build-dir=/Users/andreypopp/Workspace/esy-ocaml/esy-ocaml-project/node_modules/.cache/_esy/store/b/esy_ocaml_project-0.1.0-e39a7653'
                   with exit code: 1
esy: error, exiting...
     Error: build failed
       While Building esy-ocaml-project@0.1.0
```
dune log:
```
# /Users/andreypopp/Workspace/esy-ocaml/esy-ocaml-project/node_modules/.cache/_esy/store/i/opam__slash__dune-1.0.0-1d39b004/bin/dune build --build-dir=/Users/andreypopp/Workspace/esy-ocaml/esy-ocaml-project/node_modules/.cache/_esy/store/b/esy_ocaml_project-0.1.0-e39a7653
# OCAMLPARAM: unset
$ /usr/bin/getconf _NPROCESSORS_ONLN > /tmp/dunecc2b0b.output
$ /Users/andreypopp/.esy/3______________________________________________________________/i/ocaml-4.2.3004-0d91caab/bin/ocamlc.opt -config > /tmp/dune8797f0.output
$ (cd /Users/andreypopp/Workspace/esy-ocaml/esy-ocaml-project/node_modules/.cache/_esy/store/b/esy_ocaml_project-0.1.0-e39a7653/default && /Users/andreypopp/.esy/3______________________________________________________________/i/ocaml-4.2.3004-0d91caab/bin/ocamldep.opt -modules -impl lib/Util.ml) > /Users/andreypopp/Workspace/esy-ocaml/esy-ocaml-project/node_modules/.cache/_esy/store/b/esy_ocaml_project-0.1.0-e39a7653/default/lib/.lib.objs/Util.ml.d
$ (cd /Users/andreypopp/Workspace/esy-ocaml/esy-ocaml-project/node_modules/.cache/_esy/store/b/esy_ocaml_project-0.1.0-e39a7653/default && /Users/andreypopp/.esy/3______________________________________________________________/i/ocaml-4.2.3004-0d91caab/bin/ocamldep.opt -modules -impl bin/hello.ml) > /Users/andreypopp/Workspace/esy-ocaml/esy-ocaml-project/node_modules/.cache/_esy/store/b/esy_ocaml_project-0.1.0-e39a7653/default/bin/.hello.eobjs/hello.ml.d
```

The project is at https://github.com/esy-ocaml/esy-ocaml-project/tree/dune but it depends on a local checkout of dune to be build (I can provide instructions if needed).

**Comment by rgrinberg** [comment]
I think our test suite misses this b/c it only sets --build-dir to a relative path. I agree that's not ideal. @diml do we have a good way to make the test run with some random path in `/tmp/` perhaps?

**Comment by andreypopp** [comment]
Another data point: the error is gone when building with ocaml@4.06.0.

**Comment by diml** [comment]
we can just do `--build-dir $PWD`

**Comment by diml** [comment]
Fixed in #982. Thanks for testing master BTW!
